at-stdout/src/main/java/io/engineblock/activitytypes/stdout/StdoutActivity.java

```diff
@@ -132,15 +132,16 @@ protected Writer createPrintWriter() {
         List<StmtDef> stmts = stmtsDocList.getStmts(tagfilter);
         if (stmts.size() > 0) {
             for (StmtDef stmt : stmts) {
+                ParsedStmt parsed = stmt.getParsed().orError();
+
+                new StringCompositor(stmt.getStmt());
+                StringCompositor compositor = new StringCompositor(stmt.getStmt());
+                Map<String, String> bindingsMap = parsed.getBindings();
                 BindingsTemplate bt = new BindingsTemplate(stmt.getBindings());
-                String statement = stmt.getStmt();
-                Objects.requireNonNull(statement);
-                if (!statement.endsWith("\n") && getParams().getOptionalBoolean("newline").orElse(true)) {
-                    statement = statement+"\n";
-                }
+                Bindings bindings = bt.resolveBindings();
+                StringBindings sb = new StringBindings(compositor, bindings);
```

Comment:
This seems to be the same construct as used before, only via a different path. Specifically, the StringCompositor is still used, and its parseSection logic is the culprit. parseSection is called unconditionally from all constructors.

I have some upstream refactoring in the works that can simplify all of this by providing a single source of parsing logic. I'll reach out to review it with you in the am if you are available.